### PR TITLE
added hdf5 python bindings

### PR DESCRIPTION
**Pull request by mklpiening**
This PR adds python bindings for reading and writing images, pointclouds and tf to seerep hdf5 files.

With pybind11, this PR also adds a new dependency to seerep-hdf5 in the form of a git submodule (as suggested by the pybind developers)

Tutorials on how to use the bindings can be found in the README.md of seerep-hdf5-py

### CODE REVIEW

**Comment by mklpiening** [comment]
@Mark-Niemeyer This PR will keep on failing to build until a new version of the seerep base docker image is released. How to proceed with this? (Builds on my machine btw - yes, even with a clean docker environment :laughing: )

**Comment by Mark-Niemeyer** [comment]
Unfortunately, it is not that easy to use a new base docker image directly in the same GitHub Action. As a workaround you can trigger the rebuild and push to the ghcr of the base image [here](https://github.com/agri-gaia/seerep/actions/workflows/buildDocker.yml). I just did that, but it fails with an [error](https://github.com/agri-gaia/seerep/actions/runs/4183151358/jobs/7247143516)...

File: seerep-hdf5/seerep-hdf5-core/include/seerep-hdf5-core/hdf5-core-general.h [hunk 1]
```diff
@@ -202,6 +216,11 @@ class Hdf5CoreGeneral
 
   inline static const std::string PROJECTNAME = "projectname";
   inline static const std::string PROJECTFRAMEID = "projectframeid";
+  inline static const std::string GEOLOC_COORDINATE_SYSTEM = "geoloc_coordinatesystem";
+  inline static const std::string GEOLOC_ELLIPSOID = "geoloc_ellipsoid";
+  inline static const std::string GEOLOC_LATITUDE = "geoloc_latitude";
+  inline static const std::string GEOLOC_LONGITUDE = "geoloc_longitude";
+  inline static const std::string GEOLOC_ALTITUDE = "geoloc_altitude";
```
Comment (Mark-Niemeyer) [inline seerep-hdf5/seerep-hdf5-core/include/seerep-hdf5-core/hdf5-core-general.h:223]: Is this a rebase error?

File: seerep-hdf5/seerep-hdf5-core/include/seerep-hdf5-core/hdf5-core-tf.h [hunk 2]
```diff
@@ -41,6 +41,9 @@ class Hdf5CoreTf : public Hdf5CoreGeneral
   inline static const std::string HDF5_GROUP_TF = "tf";
 
   inline static const std::string SIZE = "size";
+
+  inline static const std::string PARENT_FRAME = "PARENT_FRAME";
+  inline static const std::string CHILD_FRAME = "CHILD_FRAME";
```
Comment (Mark-Niemeyer) [inline seerep-hdf5/seerep-hdf5-core/include/seerep-hdf5-core/hdf5-core-tf.h:46]: if you introduce these, also use them in `hdf5-core-tf.cpp` please

**Review by Mark-Niemeyer — CHANGES_REQUESTED** [review]

File: seerep-hdf5/seerep-hdf5-core/include/seerep-hdf5-core/hdf5-core-general.h (hunk 1 above)
Comment (mklpiening) [reply seerep-hdf5/seerep-hdf5-core/include/seerep-hdf5-core/hdf5-core-general.h:223]: i thinks so. im gonna change it right away :)

File: seerep-hdf5/seerep-hdf5-core/include/seerep-hdf5-core/hdf5-core-tf.h (hunk 2 above)
Comment (mklpiening) [reply seerep-hdf5/seerep-hdf5-core/include/seerep-hdf5-core/hdf5-core-tf.h:46]: I now changed all sections with hard coded "PARENT_FRAME" and "CHILD_FRAME" strings to use these constants.

**Comment by mklpiening** [comment]
please wait with merging this PR. i came across an issue with my current installation path of the python library and i am currently fixing it.

**Review by Mark-Niemeyer — APPROVED** [review]

**Pull request merged**